### ISSUE TITLE
FR: Sidechain - allow input device selection for voice activity detector

### ISSUE THREAD

**Issue by AtmanActive** [mumble-voip/mumble#6183]
### Context

Voice activated, non-push-to-talk usage on SPEAKERS (not headphones).

### Description

Right now, the sound input device is one and the same for both Voice-Activated-audio-gate detector and the main signal. There is a use case, which I'm describing below, that would need these two to be separate, like on a [sidechain compressor](https://en.wikipedia.org/wiki/Dynamic_range_compression#Side-chaining), so I can set the detector input to one device (body vibration detector) while setting the main signal audio device to another (main microphone).

### Mumble component

Client

### OS-specific?

No

### Additional information

Use case: this is about my quest to find a way to be able to play multiplayer games with my friends without having to wear headphones or use push-to-talk. 

With the use of push-to-talk, Mumble can already do this by using the attenuation feature which correctly attenuates my speakers when I'm talking or when someone else is talking.

But using push-to-talk is cumbersome and voice-activated mode can't be used because it would activate on game sounds coming out of my speakers. So I was searching for another solution.

I found voice-vibration detector hardware device here:

[Knowles V2S200D Voice Vibration Sensor](https://www.knowles.com/V2S)

This device was developed exactly for this purpose where a vibration detector can detect/amplify/ingest a signal when my voice chords are in use. Of course, this signal is a low frequency rumble and it is unusable for voice transmission. But, if it could be paired with the Mumble's Voice-Activated-audio-gate's DETECTOR input, then it could work as a reliable gate-open-close signal while the main audio signal would still be coming from my main microphone. In this way, my body, my vocal-chords-vibration would effectively be a super-accurate push-to-talk without me having to press anything and despite my speakers blaring other sounds which would reach my main microphone but would NOT reach the voice vibration sensor.

An alternative would be for me to find (or develop) a separate input-audio-gate-to-keypress software for windows which could listen to voice-vibration-sensor input and issue global keypresses to Mumble which could then do push-to-talk depending on those keypresses.

So, basically, this is organic, body activated push to talk, but right now Mumble is not ready for such a thing as it only has one audio device input for both gate-detector and gate-main signals.

Thanks.

**Comment by Krzmbrzl** [comment]
This seems quite specific and unlikely to be applicable to a wider audience, so I don't think this will be implemented by one of the core devs.
It seems like a cool idea though.

I think this would be a great fit to be implemented as a Mumble plugin. Plugin's can overwrite the transmission mode, so if you set it to PTT, a plugin can cause your voice to start to transmit and then as soon as the plugin stops the overwrite, the transmission will be stopped since there is no PTT button that is actually being pressed.

That way, you'd have to interface with your device inside the plugin and also code in the detection algorithm with which you want to detect whether or not a given input signal is enough to trigger the PTT effect.
